### PR TITLE
Design

### PR DESCRIPTION
**Pull request by andreivcodes**
styling and layout according to the design at https://www.figma.com/design/bLEBwz5KSP6CNRZ2ooyJto/Light-wallet-files?node-id=0-1&t=ZOgYNwfzZScZwOrO-0

### CODE REVIEW

**Comment by brusherru** [comment]
It breaks up the mobile version.

Needs to be fixed and tuned for mobile.
Left comprehensive comments in Slack, here are some screenshots for the protocol.

image
image
image
image
image

Does not check accounts with transactions and rewards, vault account with unlocked balance, and how it works on desktop.

**Comment by pigmej** [comment]
I dislike it, honestly. It is way too wide, and functionality is affected by it.

Smidgh is also I guess not defined shortcut too.

**Comment by dioexul** [comment]
I would prefer not to introduce that many changes before the launch. The main idea was to change colours for existing components, not to change components itself (this requires too much validation).

**Comment by andreivcodes** [comment]
Screenshot 2024-08-06 at 2 09 17 AM
Screenshot 2024-08-06 at 2 09 23 AM
Screenshot 2024-08-06 at 2 09 32 AM
Screenshot 2024-08-06 at 2 10 18 AM
Screenshot 2024-08-06 at 2 10 34 AM


Updated with fixes for mobile

**Comment by brusherru** [comment]
Well, it's better, but still:
1. Horizontal scroll on the unlock page.
    The fact that some pages still has this problem, and some not means that something implemented wrong.
    It should be part of basic template / components.

2. Transaction list still has excessive padding
    image

3. Regression: Vault account lost the feature that shows unlocked amount (https://github.com/spacemeshos/smapp-lite/pull/46)

4. I don't like that modals now lost the "darkening" of the underlaying content (or this darkening is almost invisible)
    image
    Also I think such a big paddings are excessive. Let's use the space wisely.

5. And I totally does not like this part:
    image
    There are two problems:
    - a lot of empty space on mobile because all of this does not fit in a single line
    - address, account name and switch, and it's balance are falling apart and seems not related
    
    For clarity, here is what we currently have:
    image
    
    All account related info is grouped in the single block

6. Rewards looks well enough, but I'd tweak padding a little bit to make it looks more accurate
image

7. Big amounts looks bad:
    image

8. Not sure that it makes sense to use a full width of the page on a desktop. It's harder to work with.
    image

Let's work on this better and without rush.
I think there is nothing critical to release the wallet as is, less "fancy" but more solid and logical.
And later release a really good GUI.

**Comment by noamnelke** [comment]
One thing I didn't see mentioned is the color choice. I think the "really dark green" and "slightly less dark, but still really dark green" don't have enough contrast between them. I had to turn up the brightness on my display because I wasn't even sure it was two different colors...

**Review by monikasmolarek — COMMENTED** [review]
I've reviewed the figma files and left some feedback there. Once the final layout established and implemented here, I will check this PR as well.

File: src/screens/createWallet/VerifyMnemonicScreen.tsx
Comment (brusherru) [inline src/screens/createWallet/VerifyMnemonicScreen.tsx:1]: I need more time on debugging it, but something is broken here.
Sometimes on clicking the tag in the bank it throws:
```
Uncaught Error: Cannot find next empty slot
```
And I cannot reproduce it in the main branch

File: src/screens/createWallet/CreateMnemonicScreen.tsx [hunk 1]
```diff
@@ -41,66 +47,127 @@ function CreateMnemonicScreen(): JSX.Element {
...
+        >
+          Generate new mnemonics
+        </Text>
+        <CardBody paddingY={10}>
+          <SimpleGrid columns={columns} spacing="0px">
+            {mnemonic.split(' ').map((word, idx) => {
+              const isLeftColumn = idx % columns === 0;
+              const isRightColumn = (idx + 1) % columns === 0;
+              const isTopRow = idx < columns;
+              const isBottomRow = idx >= mnemonic.split(' ').length - columns;
```
Comment (brusherru) [inline src/screens/createWallet/CreateMnemonicScreen.tsx:101]: So complicated stuff all around rendering lines...

Wouldn't grid + gap + bg color be enough?
image
image

File: src/components/CreateAccountModal.tsx [hunk 2]
```diff
@@ -162,7 +164,6 @@ function CreateAccountModal({
             />
             <FormInput
               label="Vesting start (layer number)"
-              inputProps={{ type: 'number' }}
```
Comment (brusherru) [inline src/components/CreateAccountModal.tsx:165]: Lost the important prop...

File: src/components/CreateAccountModal.tsx [hunk 3]
```diff
@@ -172,7 +173,6 @@ function CreateAccountModal({
             />
             <FormInput
               label="Vesting end (layer number)"
-              inputProps={{ type: 'number' }}
```
Comment (brusherru) [inline src/components/CreateAccountModal.tsx:175]: Lost the important prop...

File: src/components/CreateAccountModal.tsx [hunk 4]
```diff
@@ -191,7 +191,6 @@ function CreateAccountModal({
             </Text>
             <FormInput
               label="Required amount of signatures"
-              inputProps={{ type: 'number' }}
```
Comment (brusherru) [inline src/components/CreateAccountModal.tsx:194]: Lost the important prop...

File: src/components/CreateAccountModal.tsx [hunk 5]
```diff
@@ -232,7 +231,6 @@ function CreateAccountModal({
...
```
Comment (brusherru) [inline src/components/CreateAccountModal.tsx:235]: Lost the important prop...

File: src/theme/button.ts [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+  variants: {
+    green,
+    ghostGreen,
+    white,
+    ghostWhite,
+    whiteModal,
+    whiteOutline,
+    danger,
+    outline,
+  },
```
Comment (brusherru) [inline src/theme/button.ts:134]: I'm not sure it is a good solution.

Chakra's theme system supposed that there is variants.
For buttons it is `solid`, `outline`, `link` and `ghost`.
And then if you want to have different styles — use `colorCheme`. It provides a variety of different colors, but for the branding we need less, so in this case it might be good to name them like `primary`, `secondary`, `danger`, and mb `primaryDark` for cases when buttons on dark bg should be different...

But here we have a mix of these, and names are not very clear when which to use. And as I said above, I don't like that all the buttons have the same accents, for example "Add party" and "Add account" looks the same, and which is even more confusing "Cancel" and "Next" the same as well :)

To sum up, I'd like to have:
1. clear naming,
2. probably separate variants and colors (if it make sense to avoid having a big amount of variants),
3. default styles for every component, so we don't need to put `variant="whiteModal"` in every component

File: src/components/FormMultiKeySelect.tsx [hunk 7]
```diff
@@ -75,13 +75,16 @@ function FormMultiKeySelect<
             onClick={() => remove(index)}
             aria-label="Remove party"
             ml={2}
+            variant="dark"
```
[the remark this answers is not in the document]
Comment (andreivcodes) [reply src/components/FormMultiKeySelect.tsx:78]: Will resolve this using ghostWhite, which is the variant with a green hover color.
This needs a design decision for more consistency across components.

File: src/components/ImportKeyPairModal.tsx [hunk 8]
```diff
@@ -115,6 +115,18 @@ function ImportKeyPairModal({
               })}
               errors={errors}
               isSubmitted={isSubmitted}
+              textareaProps={{
+                border: '1px',
+                borderRadius: 'md',
```
[the remark this answers is not in the document]
Comment (andreivcodes) [reply src/components/ImportKeyPairModal.tsx:120]: Textareas border radius should be smaller, discussed it with Maya.
Height set to 7em, to account for padding.

File: src/components/KeyManager.tsx [hunk 9]
```diff
@@ -176,26 +175,38 @@ function KeyManager({ isOpen, onClose }: KeyManagerProps): JSX.Element {
               </TabList>
               <TabPanels display="flex" flex={1} flexDir="column" fontSize="sm">
                 <TabPanel display="flex" flexDir="column">
-                  <ButtonGroup size="sm" spacing={2} mb={4}>
-                    <Button onClick={createKeyPairModal.onOpen}>
+                  <Flex mb={4} gap={2} flexDir={['column', 'row']}>
```
[the remark this answers is not in the document]
Comment (andreivcodes) [reply src/components/KeyManager.tsx:178]: Chakra seems very opinionated about this, making it a ButtonGroup adds a left margin to all the other buttons except the first.

File: src/theme.ts [hunk 10]
```diff
@@ -1,10 +1,66 @@
...
+    100: '#34E596',
+    200: '#2ECB86',
+    300: '#28B175',
+    400: '#229764',
+    500: '#1C7D54',
+    600: '#166343',
+    700: '#104933',
+    800: '#0A2F22',
+    850: '#091F19',
+    900: '#061A14',
```
[the remark this answers is not in the document]
Comment (andreivcodes) [reply src/theme.ts:31]: some values are used in checkboxes and tabs, will switch them to the brand color palette and remove this

File: src/screens/createWallet/VerifyMnemonicScreen.tsx [hunk 11]
```diff
@@ -276,11 +359,8 @@ function DroppableBox({
   return (
     <Box
       ref={dropRef}
-      borderRadius="md"
-      borderWidth={1}
-      borderColor="gray.200"
-      bg={isOver ? 'blackAlpha.700' : 'blackAlpha.50'}
-      p={hasWordInside ? 0 : 2}
+      bg={hasWordInside ? '#0B221C' : 'brand.darkGreen'}
```
Comment (brusherru) [inline src/screens/createWallet/VerifyMnemonicScreen.tsx:362]: Btw, now slot does not react when you drag a word over it...
Please bring back the usage of `isOver` and ensure that the color is differ from bg...

**Comment by brusherru** [comment]
I still dislike it, but since we really need to officially release it and we need to have it "branded" — I'm merging it.
Hopefully all defects (design & code) will be fixed very soon.
